pkg/metrics/reporter.go

```diff
@@ -54,71 +71,71 @@ func ClusterID() uuid.UUID {
 }
 
 // ReportInstallationStarted reports that the installation has started.
-func ReportInstallationStarted(ctx context.Context) {
-	Send(ctx, InstallationStarted{
+func ReportInstallationStarted(ctx context.Context, license *kotsv1beta1.License) {
+	Send(ctx, BaseURL(license), InstallationStarted{
 		ClusterID:  ClusterID(),
 		Version:    defaults.Version,
 		Flags:      strings.Join(os.Args[1:], " "),
 		BinaryName: defaults.BinaryName(),
 		Type:       "centralized",
-		LicenseID:  LicenseID(),
+		LicenseID:  LicenseID(license),
 	})
 }
 
 // ReportInstallationSucceeded reports that the installation has succeeded.
-func ReportInstallationSucceeded(ctx context.Context) {
-	Send(ctx, InstallationSucceeded{ClusterID: ClusterID()})
+func ReportInstallationSucceeded(ctx context.Context, license *kotsv1beta1.License) {
+	Send(ctx, BaseURL(license), InstallationSucceeded{ClusterID: ClusterID()})
 }
 
 // ReportInstallationFailed reports that the installation has failed.
-func ReportInstallationFailed(ctx context.Context, err error) {
-	Send(ctx, InstallationFailed{ClusterID(), err.Error()})
+func ReportInstallationFailed(ctx context.Context, license *kotsv1beta1.License, err error) {
+	Send(ctx, BaseURL(license), InstallationFailed{ClusterID(), err.Error()})
 }
 
 // ReportJoinStarted reports that a join has started.
-func ReportJoinStarted(ctx context.Context, clusterID uuid.UUID) {
+func ReportJoinStarted(c *cli.Context, baseURL string, clusterID uuid.UUID) {
 	hostname, err := os.Hostname()
 	if err != nil {
 		logrus.Warnf("unable to get hostname: %s", err)
 		hostname = "unknown"
 	}
-	Send(ctx, JoinStarted{clusterID, hostname})
+	Send(c.Context, baseURL, JoinStarted{clusterID, hostname})
 }
 
 // ReportJoinSucceeded reports that a join has finished successfully.
-func ReportJoinSucceeded(ctx context.Context, clusterID uuid.UUID) {
+func ReportJoinSucceeded(c *cli.Context, baseURL string, clusterID uuid.UUID) {
 	hostname, err := os.Hostname()
 	if err != nil {
 		logrus.Warnf("unable to get hostname: %s", err)
 		hostname = "unknown"
 	}
-	Send(ctx, JoinSucceeded{clusterID, hostname})
+	Send(c.Context, baseURL, JoinSucceeded{clusterID, hostname})
 }
 
 // ReportJoinFailed reports that a join has failed.
-func ReportJoinFailed(ctx context.Context, clusterID uuid.UUID, exterr error) {
+func ReportJoinFailed(c *cli.Context, baseURL string, clusterID uuid.UUID, exterr error) {
```

Comment:
why require both the CLI context and the baseurl?

it seems like you only need one

Reply:
I mean you're not using c for anything besides c.Context

Reply:
ah i didn't realize that. i'll update, thanks!